src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java

```diff
@@ -0,0 +1,328 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Custom resource pack that auto overwrites the default one FML generates,
+ * allows for auto genning common models like cubes and sprite based items.
+ *
+ * @author darkevilmac (Benjamin K)
+ * @see IResourceHolder for custom texture locations for items and blocks.
+ */
+public class ConcreteResourcePack extends AbstractResourcePack implements IResourceManagerReloadListener {
+
+	private static final Logger LOG = LogManager.getLogger("Concrete");
+
+	private static Accessor<List<IResourcePack>> resourcePackList = Accessors.findField(FMLClientHandler.class, "resourcePackList");
+
+	private static Accessor<File> resourcePackFile = Accessors.findField(AbstractResourcePack.class, "field_110597_b", "resourcePackFile");
+	private static Accessor<IResourcePack> legacyPack = Accessors.findField(LegacyV2Adapter.class, "field_191383_a", "pack");
+	private static Accessor<Map<String, FallbackResourceManager>> domainResourceManagers = Accessors.findField(SimpleReloadableResourceManager.class, "field_110548_a", "domainResourceManagers");
+	private static Accessor<List<IResourcePack>> resourcePacks = Accessors.findField(FallbackResourceManager.class, "field_110540_a", "resourcePacks");
+
+	private static Invoker hasResourceName = Invokers.findMethod(AbstractResourcePack.class, null, new String[]{"func_110593_b", "hasResourceName"}, String.class);
+	private static Invoker getInputStreamByName = Invokers.findMethod(AbstractResourcePack.class, null, new String[]{"func_110591_a", "getInputStreamByName"}, String.class);
+
+	private static String SIMPLE_BLOCK_MODEL;
+	private static String SIMPLE_ITEM_MODEL;
+	private static String SIMPLE_BLOCK_STATE;
+
+	static {
+		try {
+			SIMPLE_BLOCK_MODEL = IOUtils.toString(ConcreteResourcePack.class.getResourceAsStream("concreteblockmodel.json"));
+			SIMPLE_ITEM_MODEL = IOUtils.toString(ConcreteResourcePack.class.getResourceAsStream("concreteitemmodel.json"));
+			SIMPLE_BLOCK_STATE = IOUtils.toString(ConcreteResourcePack.class.getResourceAsStream("concreteblockstate.json"));
+		} catch (IOException e) {
+			LOG.error("Caught IOException loading simple models, things will not definitely not work.", e);
+		}
+	}
+
+	public AbstractResourcePack realResourcePack;
+	public String modID;
+	public HashMap<String, String> cache;
+
+	/**
+	 * Create a ConcreteResourcePack for the specified mod, SimplePacks will auto gen simplemodels if you lack a blockstate or model file.
+	 * If the applicable file exists it will simply default to it.
+	 *
+	 * @param modID the id of the mod you want a simple pack for.
+	 */
+	public ConcreteResourcePack(String modID) {
+		super(getPackFileByModID(modID));
+		this.modID = modID;
+		this.cache = Maps.newHashMap();
+
+		// Obtain the real resource pack generated by forge.
```

Comment:
See below, we still need it for obtaining resources like already generate block models to automatically use as defaults for item models.